pkg/cmd/miscellaneous.go

```diff
@@ -165,24 +163,31 @@ func getSeedNamespaceNameForShoot(shootName string) (namespaceSeed string) {
 	return shoot.Status.TechnicalID
 }
 
-//getProjectObject returns the Project for Shoot
-func getProjectObject() (*v1beta1.Project, error) {
+//GetProjectObject returns the Project for Shoot
+func GetProjectObject(targetReader TargetReader, name string) (*v1beta1.Project, error) {
 	var target Target
 	ReadTarget(pathTarget, &target)
+
 	Client, err := target.K8SClientToKind("garden")
 	checkError(err)
-
-	if isTargeted("project") {
-		projectName, err := getTargetName("project")
+	var projectName string
+	if IsTargeted(targetReader, "project") {
+		if name == "" {
+			projectName, err = GetTargetName(targetReader, "project")
+		} else {
+			projectName = name
+		}
+		checkError(err)
+		project, err := Client.GardenerV1beta1().Projects().Get(projectName, metav1.GetOptions{})
 		checkError(err)
-		return Client.GardenerV1beta1().Projects().Get(projectName, metav1.GetOptions{})
-	} else if isTargeted("seed", "shoot") {
+		return project, nil
```

Comment:
```go
project, err := Client.GardenerV1beta1().Projects().Get(projectName, metav1.GetOptions{})
checkError(err)
return project, nil
```

Sorry this does not make much sense. What is then the purpose having the error in the return signature of the function?
I still dislike the `checkError(error)` pattern that is used across the project. I guess this also makes it harder to unit test such functions that just call process.exit instead of returning the error.

Reply:
done, 
the error return handle in checkError(err) in here, it print `err` and os.exit(1).  let me change it back. 

Yes, its true, the unit test only responds the err return like `return errors.New` or others , `checkError(error)`  which only print the `err` and ox.exit(1)

Normally I use `return errors.New` when needed in unit case `Expect`.   no special reason for I use `checkError(error)` .   I will take a look at whether to make it return `error`and os.exit  in the future.